modules/ROOT/pages/authentication-authorization/built-in-roles.adoc

```diff
@@ -1,68 +1,76 @@
 
-Neo4j provides built-in roles with default privileges. The built-in roles and the default privileges are:
+== Introduction
+
+Neo4j provides a set of built-in roles that can be used to control access to the database.
+The built-in roles are hierarchical, with the `PUBLIC` role at the bottom and the `admin` role at the top.
+The `PUBLIC` role is the default role for all users.
+It does not give any rights or capabilities regarding the data, not even read privileges.
+The `admin` role is the top-most role and has all privileges.
+
+A user may have more than one assigned role, and the union of these determines what action(s) on the data may be undertaken by the user.
+For instance, a user assigned to the `reader` role can execute procedures, because all users are also assigned to the `PUBLIC` role, which enables that capability.
+
+The built-in roles have the following default privileges:
 
 [.compact]
-`PUBLIC`::
+<<access-control-built-in-roles-public, `PUBLIC`>>::
 * Access to the home database.
-* Allows executing procedures with the users own privileges.
-* Allows executing user-defined functions with the users own privileges.
-`reader`::
+* Allows executing procedures with the users' own privileges.
+* Allows executing user-defined functions with the users' own privileges.
+<<access-control-built-in-roles-reader, `reader`>>::
 * Access to all databases.
 * Traverse and read on the data graph (all nodes, relationships, properties).
 * Show indexes and constraints along with any other future schema constructs.
-`editor`::
+<<access-control-built-in-roles-editor, `editor`>>::
 * Access to all databases.
 * Traverse, read, and write on the data graph.
-* Write access limited to creating and changing *existing* property keys, node labels, and relationship types of the graph.
+* Write access, limited to creating and changing *existing* property keys, node labels, and relationship types of the graph.
 In other words, the `editor` role cannot add to the schema but can only make changes to already existing objects.
 * Show indexes and constraints along with any other future schema constructs.
-`publisher`::
+<<access-control-built-in-roles-publisher, `publisher`>>::
 * Access to all databases.
 * Traverse, read, and write on the data graph.
 * Show indexes and constraints along with any other future schema constructs.
-`architect`::
+<<access-control-built-in-roles-architect, `architect`>>::
 * Access to all databases.
 * Traverse, read, and write on the data graph.
 * Create/drop/show indexes and constraints along with any other future schema constructs.
-`admin`::
+<<access-control-built-in-roles-admin, `admin`>>::
 * Access to all databases.
 * Traverse, read, and write on the data graph.
 * Create/drop/show indexes and constraints along with any other future schema constructs.
-* Allows executing procedures with the users own privileges or boosted privileges.
+* Allows executing procedures with the users' own privileges or boosted privileges.
```

Comment:
I know this said `own privileges or boosted privileges` before as well, but it's not like they get to choose? the privileges will be boosted for the procedure or user-defined function executions if a user has the default admin role

Reply:
I am trying to understand what that means in the context of the admin role. The admin role should have all the privileges anyway.

Reply:
I think the sentence for admin could be just
```
* Allows executing procedures using boosted privileges.
```
or something (I want to imply both allowed execute and that the execution will be boosted since we did split the two privileges). Same goes for the line about user-defined functions below.

Unless someone removes the read/write privileges (or adds denies) from admin then yes it should have all the privileges and the boosting of the procedure execution wouldn't matter.

Reply:
"* Allows executing procedures using boosted privileges." sounds much better to me. Thanks!